src/js/datamaps.js

```diff
@@ -347,27 +361,52 @@
         x = center[0] - xOffset;
         y = center[1] + yOffset;
 
-        var smallStateIndex = ["VT", "NH", "MA", "RI", "CT", "NJ", "DE", "MD", "DC"].indexOf(d.id);
-        if ( smallStateIndex > -1) {
-          var yStart = labelStartCoodinates[1];
-          x = labelStartCoodinates[0];
-          y = yStart + (smallStateIndex * (2+ (options.fontSize || 12)));
+        var smallSubunitIndex = options.smallSubunits.indexOf(d.id);
+        if ( smallSubunitIndex > -1) {
+          var yStart = labelStartCoordinates[1];
+          x = labelStartCoordinates[0];
+          y = yStart + (smallSubunitIndex * (options.labelSpacing + options.fontSize));
           layer.append("line")
             .attr("x1", x - 3)
             .attr("y1", y - 5)
             .attr("x2", center[0])
             .attr("y2", center[1])
-            .style("stroke", options.labelColor || "#000")
-            .style("stroke-width", options.lineWidth || 1)
+            .style("stroke", options.lineColor)
+            .style("stroke-width", options.lineWidth)
         }
 
-        layer.append("text")
-          .attr("x", x)
-          .attr("y", y)
-          .style("font-size", (options.fontSize || 10) + 'px')
-          .style("font-family", options.fontFamily || "Verdana")
-          .style("fill", options.labelColor || "#000")
-          .text( d.id );
+        if (!options.smallSubunitsOnly || (options.smallSubunitsOnly && smallSubunitIndex > -1) ) {
+          layer.append("text")
+            .attr("x", x)
+            .attr("y", y)
+            .style("font-size", options.fontSize + 'px')
+            .style("font-family", options.fontFamily)
+            .style("fill", options.labelColor)
+            .text(d.id)
+            .on("click", function(data, idx) {
+              var event = document.createEvent('Event');
+              event.initEvent("click", true, true);
+              subunit.dispatchEvent(event);
+            })
+            .on("mouseover", function(data, idx) {
```

Comment:
Added examples.  Also if I understand you are asking about line 378, what that was trying to do was:

Add the subunit's label to the map if it is:
a) not smallSubunitsOnly (All subunits labels show)
OR
b) it is smallSubunitsOnly, but this specific subunit exists in the subunit list to show